spec/manageiq/cloud_watch_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'cloudwatchlogger'
+require 'manageiq/loggers/cloud_watch'
+
+describe ManageIQ::Loggers::CloudWatch do
+  it "unconfigured returns a Container logger" do
+    expect(described_class.new).to be_kind_of(ManageIQ::Loggers::Container)
+  end
+
+  context "configured" do
+    around do |example|
+      ENV["CW_AWS_ACCESS_KEY_ID"] = "test"
+      ENV["CW_AWS_SECRET_ACCESS_KEY"] = "test"
+      ENV["CLOUD_WATCH_LOG_GROUP"] = "test"
+
+      example.run
+
+      ENV.delete("CW_AWS_ACCESS_KEY_ID")
+      ENV.delete("CW_AWS_SECRET_ACCESS_KEY")
+      ENV.delete("CLOUD_WATCH_LOG_GROUP")
+    end
+
+    before do
+      allow(CloudWatchLogger::Client::AWS_SDK::DeliveryThreadManager).to receive(:new).and_return(double("CloudWatchLogger::Client::AWS_SDK::DeliveryThreadManager", :deliver => nil))
+      allow(File).to receive(:exist?).and_call_original
+      allow(File).to receive(:read).and_call_original
```

Comment:
Do we need these as well as the expects below? Or can we make them all `allow`s to simplify this. They are part of the context in my opinion, rather than the test so I think allow should be alright if it simplifies things.

Reply:
I removed those `allow`s, they were a side effect of something not being stubbed early on.

I'd prefer to `expect` these things to happen.  If something is changed in the future, the `expect` will fail and we can correct it so that we avoid connecting to AWS in tests.